server/lib/src/valueset/apppwd.rs

```diff
@@ -0,0 +1,210 @@
+use crate::be::dbvalue::{DbValueApplicationPassword, DbValueSetV2};
+use crate::credential::{apppwd::ApplicationPassword, Password};
+use crate::prelude::*;
+use crate::repl::proto::{ReplApplicationPassword, ReplAttrV1};
+use crate::schema::SchemaAttribute;
+use std::collections::btree_map::Entry as BTreeEntry;
+use std::collections::BTreeMap;
+
+#[derive(Debug, Clone)]
+pub struct ValueSetApplicationPassword {
+    map: BTreeMap<Uuid, ApplicationPassword>,
+}
+
+impl ValueSetApplicationPassword {
```

Comment:
I think you need to refer to the design doc 

https://github.com/kanidm/kanidm/blob/master/book/src/developers/designs/application_passwords.md#accounts

Because of the structure you have here, you end up with code like https://github.com/kanidm/kanidm/pull/2578/files#diff-c01cb17e7b8ec9eef1cf7458a914cb61c72427e4377697d916781127a71c2942R714 which has to iterate over every possible password in the account, but in the design process, we specifically addressed this and said "lets make this structure have applicationUuid on the outside, then a multi-value structure nested inside". That way we optimise for the common path, and we get a nice, easy, common group and way to delete refered application passwords. It's a pain, but it means you have to change some of your logic and structure here, but it'll end up forcing the rest of the implementation to be nicer and map to the problem space better.

Reply:
Done, application's uuid is the map index now.